### ISSUE TITLE
Revise CenterCrop

### ISSUE THREAD

**Issue by jcohenadad** [sct-pipeline/contrast-agnostic-softseg-spinalcord#41]
The CenterCrop might be problematic for datasets where:
- the spinal cord is absolutely not at the center (eg the edge)
- there is more than 288 axial slices, because of this hardcoded param:
https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/1976cec36162f8dd1ce200cee13a3bf5493cbd9c/config_templates/meanGT_soft.json#L126

Suggestions:
- For updated dataset, find max(Z)-- or maybe not required because at inference, the CenterCrop parameter can be removed (to make sure the ENTIRE image is segmented)
- Compare with an approach where the crop is performed slice-wise along the centerline as done in [sct_deepseg_sc](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/1b5c49c3904dffe63546adbadc47759a53fbc8ca/spinalcordtoolbox/deepseg_/sc.py#L138)

**Comment by jcohenadad** [comment]
@ebadrian @louisfb01 pls see how this was done for sct_deepseg_sc and sct_deepseg_gm -- document everything you find (link to training script, preprocessing, etc)

**Comment by ebadrian** [comment]
@louisfb01 @jcohenadad Here are the script lines related to the cropping strategy for the different version of **sct_deepseg**:
- [Crop size in sct_deepseg_sc](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/deepseg_/sc.py#L458) 
- [Crop size in sct_deepseg_gm](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/deepseg_/gm.py#L191)

**Comment by naga-karthik** [comment]
Based on the current version of the model, here are the updates: 

* Cropping is still done but the crop patch size is updated to `64x192x320`. Larger RL and AP crop sizes ensure that we deal with cases with high curvature. As for SC not being in the center --> this could still be a problem, but aggressive data augmentation tries to deal with this. We also had a decent result on a scoliotic spine segmentation (result shared on slack)
* The `320` is the max size of all images in the training set in S-I direction so there is no cropping here

> For updated dataset, find max(Z)-- or maybe not required because at inference, the CenterCrop parameter can be removed (to make sure the ENTIRE image is segmented)

Yes, for inference, we're currently leaving the Z to be `-1` meaning that the model takes the original size in the S-I dim. (whatever it might be). Removing CenterCrop did not result as good results as the one with cropping. The errors were mainly spurious segmentations outsdie the SC (which can be removed in postprocessing)

Based on these (final) updates, closing this issue
